# genome_assembly_annotation

## Assembly 

1) quality control of the reads PacBio, Illumina, and RNAseq  using Fastqc and MUltiqc
2) KMer counting  using Jellyfish
3) assemblies of PAcbio reads using FLye and canu. transcriptome assemlby using Trinity
4) Polishing of assembles using Bowtie2 , pilon
5) evaluation of assemblies quality using Busco, Quast and Merqury
6) comparison of Flye and Canu against each other and the refrence using nucmer. Plot generated using mummer

## Annotation
1) Automated TE annotation and classification using EDTA 
2) Dating and phylogenetic of TE. Software clustal-omega  and FastTree  were used
3) Annotation of protein-coding sequences. Softawre genepredict was used
4) Quality assessment. Used Busco and UNiprot database
5) Comparative genomics. SOftware GENESPACE, OrthoFinder  was used. Dotplots  and riparian plot were generated.
